provd/cmd/provd/main_test.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"errors"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+type myApp struct {
+	done chan struct{}
+
+	runError         bool
+	usageErrorReturn bool
+	hupReturn        bool
+}
+
+func (a *myApp) Run() error {
+	<-a.done
+	if a.runError {
+		return errors.New("Error requested")
```

Comment:
A leftover error starting with a capital letter (yes, I know, this is on me :) ok, diminishing factor or excuse: this is only for the tests :p)